### ISSUE TITLE
bug(Table with Drag and Drop): Table with Drag and drop incorrect indexes when scrolling with overflow

### ISSUE THREAD

**Issue by Coconaut5** [angular/components#27691]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When using the material table (https://material.angular.io/components/table/overview) with the drag and drop cdk (https://material.angular.io/cdk/drag-drop/overview) there seems to be an issue in the indexes that the CdkDragDrop event emits. 

This issue only seems to happen if there is a bigger amount of items and you have overflow on any container wrapping the table. 

Locally I had issues where the previous and current index swapped values. So what was supposed to be the currentIndex was the previousIndex and vice versa. 

In the StackBlitz you can see the error where at a certain point it doesn't want to drag further down. This error only seems to happen when the overflow is on the table or a container wrapping it. But if the entire page is scrollable and the table is the full length it seems to work fine.

I also have this issue inside of the Dialog Component that can also be seen in the StackBlitz. I assume it is the same issue where the overflow cannot be on the Dialog. 

sidenote: Locally i have also experienced issues where it doesnt let me drop to any items below a certain point in the container. 

### Reproduction

StackBlitz link: https://stackblitz.com/edit/cpg8gf?file=src%2Fapp%2Ftable%2Ftable.component.ts
Steps to reproduce:
1. Pick up any item in the table from the first few rows
2. Scroll down to the last few rows and try to drop it there

### Expected Behavior

To be able to drag items freely up and down where the indexes line up properly with where it has been dropped

### Actual Behavior

Indexes are incorrect and the item doesn't drop at intended position

### Environment

- Angular: 16.1.0
- CDK/Material: 16.20.0
- Browser(s): Google Chrome, Microsoft Edge
- Operating System (e.g. Windows, macOS, Ubuntu): linux x64

**Comment by crisbeto** [comment]
It's because you need to tell the drag&drop module which parent element is scrollable using the `cdkScrollable` directive. Here's a working version of your example: https://stackblitz.com/edit/cpg8gf-pxxweh?file=src%2Fapp%2Ftable%2Ftable.component.html

**Comment by Coconaut5** [comment]
@crisbeto Thank you very much! This solved it! 

As a sidenote the issue I noticed where indexes would swap place and give incorrect values that I had locally was still there after adding cdkScrollable. However that came from a css class I had altered myself:

.cdk-drag-placeholder {
	position: absolute !important;
}

I guess it makes sense that absolute on the element could potentially interfere with the drag and drop.
